File: DebraTabs/DebraTabs/ScheduleItemType.swift
//
//  ScheduleItemType.swift
//  DebraTabs
//
//  Created by Spiro Metaxas on 3/1/17.
//  Copyright © 2017 Spiro Metaxas. All rights reserved.
//

import Foundation

enum ScheduleItemType {
    case None
    case Medication
    case GlucoseLevel
    case Exercise
    case Eating
    
    static func getJsonCode(type:ScheduleItemType)->String {
        if JSONProtocolNames.scheduleItemTypeMap[type] != nil {
            return JSONProtocolNames.scheduleItemTypeMap[type]!
        } else {
            return JSONProtocolNames.none
        }
    }
    
    static func getScheduleItemType(code:String)->ScheduleItemType {
        for (type, jsonCode) in JSONProtocolNames.scheduleItemTypeMap {
            if code == jsonCode {
                return type
            }
        }
        return .None
    }
}

File: dev/src/model/ScheduleItemType.java
package model;

public enum ScheduleItemType {
	EATING("eating"), EXERCISE("exercise"), MEDICATION("medication"), BG_MEASUREMENT("blood_glucose_measurement");
	
	private String name;
	
	private ScheduleItemType(String name) {
		this.name = name;
	}
	
	@Override
	public String toString() {
		return name;
	}
	
	public int scoreBaseByScheduleItemType() {
		switch(this) {
			case EATING:
				return 50;
			case EXERCISE:
				return 100;
			case MEDICATION:
				return 20;
			case BG_MEASUREMENT:
				return 200;
			default:
				throw new IllegalStateException();
		}
	}
}
